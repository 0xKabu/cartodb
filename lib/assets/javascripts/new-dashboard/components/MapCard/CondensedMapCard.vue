<template>
  <a :href="vizUrl"
     class="row"
     :class="{
       'row--selected': isSelected,
       'row--quick-actions-open': areQuickActionsOpen,
       'row--no-hover': !activeHover,
       'row--can-hover': canHover
     }"
     @click="onClick">

    <div class="cell cell--thumbnail">
      <div class="card-media">
        <img width="48" height="48" class="cell--thumbnail__img" :src="mapThumbnailUrl" @error="onThumbnailError" v-if="!isThumbnailErrored"/>
      </div>
      <span class="checkbox cell--thumbnail__checkbox" @mouseover="mouseOverChildElement" @mouseleave="mouseOutChildElement">
        <input class="checkbox-input" :checked="isSelected" @click.stop="toggleSelection" type="checkBox">
        <span class="checkbox-decoration">
          <img svg-inline src="../../assets/icons/common/checkbox.svg">
        </span>
      </span>
    </div>

    <div class="cell cell--map-name">
      <div class="title-container">
        <h3 class="text is-caption is-txtGrey u-ellipsis cell--map-name__text">{{ map.name }}</h3>

        <span v-if="showInteractiveElements" class="card-favorite" :class="{'is-favorite': map.liked, 'favorite-overflow': titleOverflow}" @click.prevent="toggleFavorite" @mouseover="mouseOverChildElement" @mouseleave="mouseOutChildElement">
            <img svg-inline src="../../assets/icons/common/favorite.svg">
        </span>
      </div>
    </div>

    <div class="cell cell--large">
      <span class="text is-small is-txtSoftGrey">{{ lastUpdated }}</span>
    </div>

    <div class="cell cell--large">
      <span class="text is-small is-txtSoftGrey">{{ $tc(`MapCard.condensedViews`, numberViews )}}</span>
    </div>

    <div class="cell cell--privacy cell--medium">
      <span class="icon icon--privacy" :class="privacyIcon"></span>
      <p class="text is-small is-txtSoftGrey">
        {{ $t(`MapCard.shared.${map.privacy}`) }}
      </p>
    </div>

    <div class="cell cell--actions" @mouseover="mouseOverChildElement" @mouseleave="mouseOutChildElement">
      <span class="quick-actions-placeholder" v-if="!showInteractiveElements"></span>
      <MapQuickActions v-if="showInteractiveElements" :map="map" @open="openQuickActions" @close="closeQuickActions" @dataChanged="onDataChanged" :hasShadow="false" />
    </div>
  </a>
</template>

<script>

import computed from './shared/computed';
import data from './shared/data';
import FeaturesDropdown from 'new-dashboard/components/Dropdowns/FeaturesDropdown';
import MapQuickActions from 'new-dashboard/components/QuickActions/MapQuickActions';
import methods from './shared/methods';
import props from './shared/props';

export default {
  name: 'CondensedMapCard',
  components: {
    MapQuickActions,
    FeaturesDropdown
  },
  props,
  data,
  computed,
  methods
};
</script>

<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.row {
  display: flex;
  align-items: center;
  width: 100%;
  height: 80px;
  padding: 0 14px;
  transition: all 0.25s cubic-bezier(0.4, 0.01, 0.165, 0.99);
  background-color: #FFF;

  .cell--thumbnail__checkbox {
    position: absolute;
    left: 22px;
    transform: translateY(250%);
    transition: all 0.25s cubic-bezier(0.4, 0.01, 0.165, 0.99);
    opacity: 0;
    pointer-events: none;
  }

  .cell {
    position: relative;
    flex-grow: 0;
    flex-shrink: 0;
    padding: 0 12px;
    text-overflow: ellipsis;
    white-space: nowrap;
  }

  .cell--large {
    width: 150px;
  }

  .cell--medium {
    width: 120px;
  }

  .cell--thumbnail {
    display: flex;
    align-items: center;
    align-self: flex-start;
    height: 100%;
    overflow: hidden;

    &__img {
      width: 48px;
      height: 48px;
    }
  }

  .cell--map-name {
    flex-grow: 1;
    flex-shrink: 1;
    min-width: 200px;

    .card-favorite {
      display: inline-block;
      margin-left: 8px;
      opacity: 0;
      vertical-align: middle;

      &.is-favorite {
        opacity: 1;

        .favorite-icon {
          stroke: #FFC300;
          fill: #FFC300;
        }

        &:hover {
          .favorite-icon {
            stroke: $primary-color;
          }
        }
      }
    }
  }

  .cell--privacy {
    display: flex;
    flex-direction: row;

    .icon {
      display: flex;
      justify-content: center;
      width: 18px;
      height: 18px;
      margin-right: 8px;
      background-repeat: no-repeat;
      background-position: center;

      &.icon--privacy {
        background-repeat: no-repeat;
        background-position: center;
      }

      &.icon--private {
        background-image: url("../../assets/icons/maps/privacy/lock.svg");
      }

      &.icon--public {
        background-image: url("../../assets/icons/maps/privacy/public.svg");
      }

      &.icon--link {
        background-image: url("../../assets/icons/maps/privacy/link.svg");
      }

      &.icon--password {
        background-image: url("../../assets/icons/maps/privacy/password.svg");
      }

      &.icon--sharedBy {
        border-radius: 2px;
        background-size: contain;
      }
    }
  }

  .cell--actions {
    visibility: hidden;

    .quick-actions-placeholder {
      display: block;
      width: 24px;
      height: 24px;
    }
  }

  &.row--selected {
    box-shadow: 0 0 0 1px $primary-color;
  }

  &:hover,
  &.row--selected {
    &.row--can-hover {
      .card-media {
        transform: translateY(-100%);
        opacity: 0;
      }

      .cell--thumbnail__checkbox {
        transform: translateY(0);
        opacity: 1;
        pointer-events: all;
      }
    }
  }

  &:hover {
    text-decoration: none;

    .cell--map-name {
      .cell--map-name__text {
        color: $primary-color;
      }

      .card-favorite {
        opacity: 1;

        .favorite-icon {
          stroke: $text-secondary-color;
        }

        &:hover {
          .favorite-icon {
            stroke: $primary-color;
          }
        }
      }
    }

    .cell--actions {
      visibility: initial;
    }
  }

  .title-container {
    display: flex;
    align-items: center;
  }

  .card-media {
    display: flex;
    position: relative;
    width: 48px;
    height: 48px;
    overflow: hidden;
    border-radius: 2px;
    background: url($assetsDir + '/images/layout/default-map-bkg.png') no-repeat center 0;
    background-size: cover;
  }
}
</style>
