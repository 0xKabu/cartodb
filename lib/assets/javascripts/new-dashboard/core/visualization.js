import PermissionModel from 'dashboard/data/permission-model';
import store from 'new-dashboard/store';

export function getURL (visualizationData) {
  const loggedUserBackboneModel = store.state.user.userModel;
  const permissionModel = new PermissionModel(
    visualizationData.permission, { configModel: store.state.config.configModel }
  );

  let id = visualizationData.id;
  let userUrl = permissionModel.owner.viewUrl();

  if (loggedUserBackboneModel &&
      loggedUserBackboneModel.id !== permissionModel.owner.get('id') &&
      permissionModel.hasAccess(loggedUserBackboneModel)) {
    userUrl = loggedUserBackboneModel.viewUrl();
    id = `${permissionModel.owner.get('username')}.${id}`;
  }

  return `${userUrl.toString()}/viz/${id}`;
};
