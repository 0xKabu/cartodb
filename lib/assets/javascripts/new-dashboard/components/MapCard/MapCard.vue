<template>
  <SimpleMapCard
    v-if="!condensed"
    :map="map"
    :isSelected="isSelected"
    :canHover="canHover"
    :selectMode="selectMode"
    @toggleSelection="toggleSelection"
    @toggleFavorite="toggleFavorite"/>
  <CondensedMapCard
    v-else
    :map="map"
    :isSelected="isSelected"
    :canHover="canHover"
    :selectMode="selectMode"
    @toggleSelection="toggleSelection"
    @toggleFavorite="toggleFavorite"/>
</template>

<script>
import CondensedMapCard from './CondensedMapCard';
import SimpleMapCard from './SimpleMapCard';

export default {
  name: 'MapCard',
  props: {
    condensed: {
      type: Boolean,
      default: false
    },
    map: Object,
    isSelected: {
      type: Boolean,
      default: false
    },
    canHover: {
      type: Boolean,
      default: true
    },
    selectMode: {
      type: Boolean,
      default: false
    }
  },
  components: {
    SimpleMapCard,
    CondensedMapCard
  },
  methods: {
    toggleSelection () {
      this.$emit('toggleSelection', {
        map: this.$props.map,
        isSelected: !this.$props.isSelected
      });
    }
  }
};
</script>
