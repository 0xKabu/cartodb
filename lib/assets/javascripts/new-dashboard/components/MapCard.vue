<template>
  <div class="grid-cell" :class="sizeClass">
    <a :href="vizUrl" class="card map-card" :class="{selected: selected, 'card--noHover': !activeHover}">
      <span class="checkbox card-select" @mouseover="mouseOverElement" @mouseleave="mouseOutOfElement">
        <input class="checkbox-input" @click="toggleSelection" type="checkBox">
        <span class="checkbox-decoration">
            <svg viewBox="0 0 12 12" class="checkbox-decorationMedia">
                <g fill="none">
                    <polyline class="checkbox-check" points="1.65093994 3.80255127 4.48919678 6.97192383 10.3794556 0.717346191"></polyline>
                </g>
            </svg>
        </span>
      </span>
      <div class="card-actions" @mouseover="mouseOverElement" @mouseleave="mouseOutOfElement">
        <span class="card-actionsSelect">
            <img src="../assets/icons/common/options.svg">
        </span>
      </div>
      <div class="card-media">
        <img :src=mapThumbnailUrl />
      </div>
      <div class="card-text">
        <h2 class="card-title title is-caption">
          {{ map.name }}&nbsp;
          <span class="card-favorite" v-bind:class="{'is-favorite': favorite}" @click.prevent="toggleFavorite" @mouseover="mouseOverElement" @mouseleave="mouseOutOfElement">
            <svg width="16" height="17" viewBox="0 0 16 17" xmlns="http://www.w3.org/2000/svg">
              <path class="favorite-icon" d="M15.44 5.46a.75.75 0 0 0-.69-.46h-4.04L8.67.92C8.42.4 7.58.4 7.33.92L5.29 5H1.25a.75.75 0 0 0-.53 1.28l3.44 3.44-1.38 4.83a.75.75 0 0 0 1.14.82L8 12.65l4.08 2.72a.75.75 0 0 0 1.14-.82l-1.38-4.83 3.44-3.44a.75.75 0 0 0 .16-.82z" stroke="#6F757B" fill="none" fill-rule="evenodd"/>
            </svg>
          </span>
        </h2>
        <p class="card-description text is-caption" v-if="map.description">{{ map.description }}</p>
        <p class="card-description text is-caption" v-else>{{ $t(`mapCard.noDescription`) }}</p>
        <ul class="card-metadata">
          <li class="card-metadataItem text is-caption">
            <span class="icon icon--privacy" :class="privacyIcon"></span>
            <p>{{ $t(`mapCard.shared.${map.privacy}`) }}</p>
          </li>
          <li class="card-metadataItem text is-caption">
            <span class="icon"><img src="../assets/icons/maps/calendar.svg"></span>
            <p>{{ lastUpdated }}</p>
          </li>
          <li class="card-metadataItem text is-caption">
            <span class="icon"><img src="../assets/icons/maps/tag.svg"></span>
            <ul class="card-tagList">
              <li v-for="(tag, index) in map.tagList" :key="tag">
                <a href="#" @mouseover="mouseOverElement" @mouseleave="mouseOutOfElement">{{ tag }}</a><span v-if="index < map.tagList.length - 1">,&#32;</span>
              </li>
              <li v-if="map.tagList.length <= 0">
                <span>{{ $t(`mapCard.noTags`) }}</span>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </a>
  </div>
</template>

<script>
import distanceInWordsStrict from 'date-fns/distance_in_words_strict';
import * as Visualization from 'new-dashboard/core/visualization';

export default {
  name: 'CardMap',
  props: {
    map: Object,
    size: {
      type: String,
      default: 'small'
    }
  },
  data: function () {
    return {
      selected: false,
      favorite: this.$props.map.favorite,
      activeHover: true,
      sizeClasses: {
        medium: 'grid-cell--col6 grid-cell--col12--mobile',
        small: 'grid-cell--col4 grid-cell--col6--tablet grid-cell--col12--mobile'
      }
    };
  },
  computed: {
    privacyIcon () {
      return `icon--${this.$props.map.privacy}`.toLowerCase();
    },
    lastUpdated () {
      return `Updated ${distanceInWordsStrict(this.$props.map.updated_at, new Date())} ago`;
    },
    sizeClass () {
      return this.sizeClasses[this.$props.size];
    },
    mapThumbnailUrl () {
      return this.$props.map.thumbnailUrl;
    },
    vizUrl () {
      return Visualization.getURL(this.$props.map);
    }
  },
  methods: {
    toggleSelection () {
      this.selected = !this.selected;
    },
    toggleFavorite () {
      this.favorite = !this.favorite;
    },
    mouseOverElement () {
      this.activeHover = false;
    },
    mouseOutOfElement () {
      this.activeHover = true;
    }
  }
};
</script>

<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.map-card {
  display: block;
  text-decoration: none;
}

.card {
  position: relative;
  height: 100%;
  transition: background 300ms cubic-bezier(0.4, 0, 0.2, 1);
  background-color: $white;

  &::after {
    content: '';
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    border: 1px solid $light-grey;
    pointer-events: none;
  }

  &:hover {
    cursor: pointer;

    &:not(.card--noHover) {
      .card-title {
        color: $primary-color;
      }
    }

    .card-select,
    .card-actions,
    .card-favorite {
      opacity: 1;
    }
  }

  &.selected {
    background-color: #F2F9FF;

    .card-actions,
    .card-select {
      opacity: 1;
    }
  }
}

.card-text {
  padding: 24px 16px;
  color: $text-color;
}

.card-media {
  display: flex;
  height: 140px;
  overflow: hidden;

  img {
    width: 100%;
    object-fit: cover;
  }
}

.card-title {
  margin-bottom: 12px;
  transition: background 300ms cubic-bezier(0.4, 0, 0.2, 1);
}

.card-description {
  display: -webkit-box;
  display: block;
  height: 48px;
  margin-bottom: 8px;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
}

.card-metadataItem {
  display: flex;
  margin-bottom: 4px;

  a {
    color: $text-color;
    text-decoration: none;
  }

  &:last-child {
    margin-bottom: 0;
  }

  a:hover {
    color: $text-color;
    text-decoration: underline;
  }

  .icon {
    width: 18px;
    height: 18px;
    margin-top: 4px;
    margin-right: 8px;

    &.icon--privacy {
      background-repeat: no-repeat;
      background-position: center;
    }

    &.icon--private {
      background-image: url("../assets/icons/maps/privacy/lock.svg");
    }

    &.icon--public {
      background-image: url("../assets/icons/maps/privacy/public.svg");
    }

    &.icon--link {
      background-image: url("../assets/icons/maps/privacy/link.svg");
    }

    &.icon--password {
      background-image: url("../assets/icons/maps/privacy/password.svg");
    }
  }
}

.card-select {
  position: absolute;
  top: 8px;
  left: 8px;
  transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1);
  border-radius: 4px;
  opacity: 0;
  background: $white;
  box-shadow: 1px 1px 1px rgba(0, 0, 0, 0.12);
  -webkit-appearance: none;
  appearance: none;
  cursor: pointer;

  &:focus {
    outline: none;
  }

  &:checked {
    opacity: 1;
    background: $primary-color;
  }
}

.card-actions {
  position: absolute;
  top: 8px;
  right: 8px;
  transition: opacity 300ms cubic-bezier(0.4, 0, 0.2, 1);
  opacity: 0;
}

.card-actionsSelect {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 24px;
  height: 24px;
  border-radius: 4px;
  background: $white;
  box-shadow: 1px 1px 1px rgba(0, 0, 0, 0.12);
}

.card-favorite {
  margin-left: 4px;
  opacity: 0;

  svg {
    transform: translateY(2px);
  }

  &:hover {
    .favorite-icon {
      stroke: $primary-color;
    }
  }

  &.is-favorite {
    opacity: 1;

    .favorite-icon {
      stroke: #FFC300;
      fill: #FFC300;
    }

    &:hover {
      .favorite-icon {
        stroke: $primary-color;
      }
    }
  }
}

.card-tagList > li {
  display: inline;
}
</style>
