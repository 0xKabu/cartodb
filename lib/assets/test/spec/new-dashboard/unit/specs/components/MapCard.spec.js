import { shallowMount, mount } from '@vue/test-utils';
import visualization from '../fixtures/visualizations';
import MapCard from 'new-dashboard/components/MapCard';

// Backbone Models
import ConfigModel from 'dashboard/data/config-model';
import UserModel from 'dashboard/data/user-model';

let $cartoModels;
let $t;

describe('MapCard.vue', () => {
  beforeEach(() => {
    const user = new UserModel({});
    const config = new ConfigModel({});
    $cartoModels = { user, config };
    $t = key => key;
  });

  it('should render correct contents', () => {
    const map = visualization.visualizations[0];
    const mapCard = shallowMount(MapCard, {
      propsData: { map },
      mocks: {
        $cartoModels,
        $t
      }
    });
    expect(mapCard).toMatchSnapshot();
  });

  it('should render description and tags', () => {
    const map = visualization.visualizations[0];
    map.description = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.';
    map.tags = ['Hi', 'Hello', 'CARTO Rules'];
    const mapCard = shallowMount(MapCard, {
      propsData: { map },
      mocks: {
        $cartoModels,
        $t
      }
    });
    expect(mapCard).toMatchSnapshot();
  });

  describe('Methods', () => {
    it('should toggle selected state', () => {
      const map = visualization.visualizations[0];
      const mapCard = shallowMount(MapCard, {
        propsData: { map },
        mocks: {
          $cartoModels,
          $t
        }
      });
      expect(mapCard.vm.selected).toBe(false);
      mapCard.vm.toggleSelection();
      expect(mapCard.vm.selected).toBe(true);
      expect(mapCard).toMatchSnapshot();
    });

    it('should toggle favorite state', () => {
      const map = visualization.visualizations[0];
      const mapCard = shallowMount(MapCard, {
        propsData: { map },
        mocks: {
          $cartoModels,
          $t
        }
      });
      expect(mapCard.vm.$props.map.liked).toBe(true);
      mapCard.vm.toggleFavorite();
      expect(mapCard.vm.$props.map.liked).toBe(false);
      expect(mapCard).toMatchSnapshot();
    });

    it('should toggle mouse hover', () => {
      const map = visualization.visualizations[0];
      const mapCard = mount(MapCard, {
        propsData: { map },
        mocks: {
          $cartoModels,
          $t
        }
      });

      expect(mapCard.vm.activeHover).toBe(true);
      mapCard.find('.card-select').trigger('mouseover');
      // mapCard.vm.mouseOverChildElement();
      expect(mapCard.vm.activeHover).toBe(false);
      expect(mapCard).toMatchSnapshot();

      // mapCard.vm.mouseOutChildElement();
      mapCard.find('.card-select').trigger('mouseleave');
      expect(mapCard.vm.activeHover).toBe(true);
      expect(mapCard).toMatchSnapshot();
    });

    it('should show thumbnail error', () => {
      const map = visualization.visualizations[0];
      const mapCard = shallowMount(MapCard, {
        propsData: { map },
        mocks: {
          $cartoModels,
          $t
        }
      });
      expect(mapCard.vm.isThumbnailErrored).toBe(false);
      mapCard.vm.onThumbnailError();
      expect(mapCard.vm.isThumbnailErrored).toBe(true);
      expect(mapCard).toMatchSnapshot();
    });
  });
});
