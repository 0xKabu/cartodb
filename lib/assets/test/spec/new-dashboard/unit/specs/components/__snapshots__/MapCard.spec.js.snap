// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapCard.vue Methods should show thumbnail error 1`] = `
<a href="https://team.carto.com/u/cillas/viz/f8e13983-bb08-4ca9-b64a-f34e76fe077a" class="card map-card">
  <div class="card-media has-error">
    <!---->
    <div class="MapCard-error"></div>
  </div> <span class="checkbox card-select"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../assets/icons/common/checkbox.svg"></span></span>
  <div class="card-actions"><span class="card-actionsSelect"><img src="../assets/icons/common/options.svg"></span></div>
  <div class="card-text">
    <div class="card-header">
      <h2 class="card-title title is-caption">
        Airbnb listings VS metro stations&nbsp;
        <span class="card-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p class="card-description text is-caption">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
    </div>
    <ul class="card-metadata">
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--private"></span>
        <p>mapCard.shared.PRIVATE</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>mapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <ul class="card-tagList">
          <li><a href="#">Hi</a><span>, </span></li>
          <li><a href="#">Hello</a><span>, </span></li>
          <li><a href="#">CARTO Rules</a>
            <!---->
          </li>
          <!---->
        </ul>
      </li>
    </ul>
  </div>
</a>
`;

exports[`MapCard.vue Methods should toggle favorite state 1`] = `
<a href="https://team.carto.com/u/cillas/viz/f8e13983-bb08-4ca9-b64a-f34e76fe077a" class="card map-card">
  <div class="card-media"><img src="http://undefined/api/v1/map/static/named/tpl_f8e13983_bb08_4ca9_b64a_f34e76fe077a/600/280.png?auth_token=feZhHEP4vJEbTbF3PEgyvA&amp;auth_token=00580a1519bad8a50dba717dc0bb82bd357084c01212ba90f6058a226457e77d">
    <!---->
  </div> <span class="checkbox card-select"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../assets/icons/common/checkbox.svg"></span></span>
  <div class="card-actions"><span class="card-actionsSelect"><img src="../assets/icons/common/options.svg"></span></div>
  <div class="card-text">
    <div class="card-header">
      <h2 class="card-title title is-caption">
        Airbnb listings VS metro stations&nbsp;
        <span class="card-favorite is-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p class="card-description text is-caption">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
    </div>
    <ul class="card-metadata">
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--private"></span>
        <p>mapCard.shared.PRIVATE</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>mapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <ul class="card-tagList">
          <li><a href="#">Hi</a><span>, </span></li>
          <li><a href="#">Hello</a><span>, </span></li>
          <li><a href="#">CARTO Rules</a>
            <!---->
          </li>
          <!---->
        </ul>
      </li>
    </ul>
  </div>
</a>
`;

exports[`MapCard.vue Methods should toggle mouse hover 1`] = `
<a href="https://team.carto.com/u/cillas/viz/f8e13983-bb08-4ca9-b64a-f34e76fe077a" class="card map-card card--noHover">
  <div class="card-media"><img src="http://undefined/api/v1/map/static/named/tpl_f8e13983_bb08_4ca9_b64a_f34e76fe077a/600/280.png?auth_token=feZhHEP4vJEbTbF3PEgyvA&amp;auth_token=00580a1519bad8a50dba717dc0bb82bd357084c01212ba90f6058a226457e77d">
    <!---->
  </div> <span class="checkbox card-select"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../assets/icons/common/checkbox.svg"></span></span>
  <div class="card-actions"><span class="card-actionsSelect"><img src="../assets/icons/common/options.svg"></span></div>
  <div class="card-text">
    <div class="card-header">
      <h2 class="card-title title is-caption">
        Airbnb listings VS metro stations&nbsp;
        <span class="card-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p class="card-description text is-caption">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
    </div>
    <ul class="card-metadata">
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--private"></span>
        <p>mapCard.shared.PRIVATE</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>mapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <ul class="card-tagList">
          <li><a href="#">Hi</a><span>, </span></li>
          <li><a href="#">Hello</a><span>, </span></li>
          <li><a href="#">CARTO Rules</a>
            <!---->
          </li>
          <!---->
        </ul>
      </li>
    </ul>
  </div>
</a>
`;

exports[`MapCard.vue Methods should toggle mouse hover 2`] = `
<a href="https://team.carto.com/u/cillas/viz/f8e13983-bb08-4ca9-b64a-f34e76fe077a" class="card map-card">
  <div class="card-media"><img src="http://undefined/api/v1/map/static/named/tpl_f8e13983_bb08_4ca9_b64a_f34e76fe077a/600/280.png?auth_token=feZhHEP4vJEbTbF3PEgyvA&amp;auth_token=00580a1519bad8a50dba717dc0bb82bd357084c01212ba90f6058a226457e77d">
    <!---->
  </div> <span class="checkbox card-select"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../assets/icons/common/checkbox.svg"></span></span>
  <div class="card-actions"><span class="card-actionsSelect"><img src="../assets/icons/common/options.svg"></span></div>
  <div class="card-text">
    <div class="card-header">
      <h2 class="card-title title is-caption">
        Airbnb listings VS metro stations&nbsp;
        <span class="card-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p class="card-description text is-caption">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
    </div>
    <ul class="card-metadata">
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--private"></span>
        <p>mapCard.shared.PRIVATE</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>mapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <ul class="card-tagList">
          <li><a href="#">Hi</a><span>, </span></li>
          <li><a href="#">Hello</a><span>, </span></li>
          <li><a href="#">CARTO Rules</a>
            <!---->
          </li>
          <!---->
        </ul>
      </li>
    </ul>
  </div>
</a>
`;

exports[`MapCard.vue Methods should toggle selected state 1`] = `
<a href="https://team.carto.com/u/cillas/viz/f8e13983-bb08-4ca9-b64a-f34e76fe077a" class="card map-card selected">
  <div class="card-media"><img src="http://undefined/api/v1/map/static/named/tpl_f8e13983_bb08_4ca9_b64a_f34e76fe077a/600/280.png?auth_token=feZhHEP4vJEbTbF3PEgyvA&amp;auth_token=00580a1519bad8a50dba717dc0bb82bd357084c01212ba90f6058a226457e77d">
    <!---->
  </div> <span class="checkbox card-select"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../assets/icons/common/checkbox.svg"></span></span>
  <div class="card-actions"><span class="card-actionsSelect"><img src="../assets/icons/common/options.svg"></span></div>
  <div class="card-text">
    <div class="card-header">
      <h2 class="card-title title is-caption">
        Airbnb listings VS metro stations&nbsp;
        <span class="card-favorite is-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p class="card-description text is-caption">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
    </div>
    <ul class="card-metadata">
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--private"></span>
        <p>mapCard.shared.PRIVATE</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>mapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <ul class="card-tagList">
          <li><a href="#">Hi</a><span>, </span></li>
          <li><a href="#">Hello</a><span>, </span></li>
          <li><a href="#">CARTO Rules</a>
            <!---->
          </li>
          <!---->
        </ul>
      </li>
    </ul>
  </div>
</a>
`;

exports[`MapCard.vue should render correct contents 1`] = `
<a href="https://team.carto.com/u/cillas/viz/f8e13983-bb08-4ca9-b64a-f34e76fe077a" class="card map-card">
  <div class="card-media"><img src="http://undefined/api/v1/map/static/named/tpl_f8e13983_bb08_4ca9_b64a_f34e76fe077a/600/280.png?auth_token=feZhHEP4vJEbTbF3PEgyvA&amp;auth_token=00580a1519bad8a50dba717dc0bb82bd357084c01212ba90f6058a226457e77d">
    <!---->
  </div> <span class="checkbox card-select"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../assets/icons/common/checkbox.svg"></span></span>
  <div class="card-actions"><span class="card-actionsSelect"><img src="../assets/icons/common/options.svg"></span></div>
  <div class="card-text">
    <div class="card-header">
      <h2 class="card-title title is-caption">
        Airbnb listings VS metro stations&nbsp;
        <span class="card-favorite is-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p class="card-description text is-caption is-txtSoftGrey">mapCard.noDescription</p>
    </div>
    <ul class="card-metadata">
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--private"></span>
        <p>mapCard.shared.PRIVATE</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>mapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <ul class="card-tagList">
          <li><span>mapCard.noTags</span></li>
        </ul>
      </li>
    </ul>
  </div>
</a>
`;

exports[`MapCard.vue should render description and tags 1`] = `
<a href="https://team.carto.com/u/cillas/viz/f8e13983-bb08-4ca9-b64a-f34e76fe077a" class="card map-card">
  <div class="card-media"><img src="http://undefined/api/v1/map/static/named/tpl_f8e13983_bb08_4ca9_b64a_f34e76fe077a/600/280.png?auth_token=feZhHEP4vJEbTbF3PEgyvA&amp;auth_token=00580a1519bad8a50dba717dc0bb82bd357084c01212ba90f6058a226457e77d">
    <!---->
  </div> <span class="checkbox card-select"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../assets/icons/common/checkbox.svg"></span></span>
  <div class="card-actions"><span class="card-actionsSelect"><img src="../assets/icons/common/options.svg"></span></div>
  <div class="card-text">
    <div class="card-header">
      <h2 class="card-title title is-caption">
        Airbnb listings VS metro stations&nbsp;
        <span class="card-favorite is-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p class="card-description text is-caption">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
    </div>
    <ul class="card-metadata">
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--private"></span>
        <p>mapCard.shared.PRIVATE</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>mapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <ul class="card-tagList">
          <li><a href="#">Hi</a><span>, </span></li>
          <li><a href="#">Hello</a><span>, </span></li>
          <li><a href="#">CARTO Rules</a>
            <!---->
          </li>
          <!---->
        </ul>
      </li>
    </ul>
  </div>
</a>
`;
